ciml/gather_results.py

```diff
@@ -532,23 +534,48 @@ def get_experiment_folder(dataset, name, data_path=None):
     return get_data_json_folder(dataset, sub_folder=name, data_path=data_path)
 
 
-def load_dataset(dataset, name, data_path=None):
-    if not data_path:
-        dataset_file = [os.path.dirname(os.path.realpath(__file__)), os.pardir]
+def load_dataset(dataset, name, data_path=None, s3=None):
+    dataset_file = get_data_path(data_path=data_path, s3=s3)
+    dataset_file.append(dataset)
+    use_s3 = (dataset_file[0] == 's3:')
+
+    if use_s3:
+        # Download as a file first. Not that efficient, but we avoid having
+        # to keep the whole dataset in memory.
+        object_key = os.sep.join(dataset_file[3:] + [name + '.npz'])
+        _, target_filename = tempfile.mkstemp(prefix=None, suffix='.npz')
+        s3.download_file(Bucket=dataset_file[2], Key=object_key,
```

Comment:
What if s3 is None here? Don't we need to initialize a client?

Reply:
There are two ways we can solve this, either fail is s3 is not passed and data_path is s3://, or else we can get an s3 client on the fly, only we don't have the s3 parameters, so we can get a vanilla one which may not be the right thing to do. So probably we should fail.

Reply:
Updated the code, throwing ValueError now.